search/util.go

```diff
@@ -144,5 +145,23 @@ type ScoreExplCorrectionCallbackFunc func(queryMatch *DocumentMatch, knnMatch *D
 type SearcherStartCallbackFn func(size uint64) error
 type SearcherEndCallbackFn func(size uint64) error
 
+// field -> term -> synonyms
+type FieldTermSynonymMap map[string]map[string][]string
+
+func (f *FieldTermSynonymMap) MergeWith(fts FieldTermSynonymMap) {
+	for field, termSynonymMap := range fts {
+		// Ensure the field exists in the receiver
+		if _, exists := (*f)[field]; !exists {
+			(*f)[field] = make(map[string][]string)
+		}
+		for term, synonyms := range termSynonymMap {
+			// Append synonyms
+			(*f)[field][term] = append((*f)[field][term], synonyms...)
```

Comment:
Doesn't look very GO idiomatic at all, with the above change this can become `f[field][term]` ?